app/dao/templates_dao.py

```diff
@@ -89,70 +95,98 @@ def dao_redact_template(template, user_id):
 
 
 def dao_get_template_by_id_and_service_id(template_id, service_id, version=None) -> Template:
-    if version is not None:
-        return TemplateHistory.query.filter_by(
-            id=template_id,
-            hidden=False,
-            service_id=service_id,
-            version=version).one()
-    return Template.query.filter_by(id=template_id, hidden=False, service_id=service_id).one()
+    if version is None:
+        stmt = select(Template).where(
+            Template.id == template_id,
+            Template.hidden.is_(False),
+            Template.service_id == service_id
+        )
+    else:
+        stmt = select(TemplateHistory).where(
+            TemplateHistory.id == template_id,
+            TemplateHistory.hidden.is_(False),
+            TemplateHistory.service_id == service_id,
+            TemplateHistory.version == version
+        )
+
+    return db.session.scalars(stmt).one()
 
 
 def dao_get_number_of_templates_by_service_id_and_name(service_id, template_name, version=None):
-    if version is not None:
-        return TemplateHistory.query.filter_by(
-            hidden=False,
-            service_id=service_id,
-            name=template_name,
-            version=version).count()
-    return Template.query.filter_by(hidden=False, service_id=service_id, name=template_name).count()
+    if version is None:
+        stmt = select(func.count()).select_from(Template).where(
+            Template.hidden.is_(False),
+            Template.service_id == service_id,
+            Template.name == template_name
+        )
+    else:
+        stmt = select(func.count()).select_from(TemplateHistory).where(
+            TemplateHistory.hidden.is_(False),
```

Comment:
Better way?

Reply:
Note sure what "better way" means? What did we agree on? Needs changes?

Reply:
No change.  is_ is the correct syntax to test booleans.